docs/source/using/experimental-features.adoc

```diff
@@ -151,3 +151,163 @@ To get the status of the DNS server:
 ----
 $ minishift dns status
 ----
+
+[[local-dns-setup-macos]]
+== Local DNS Set-up for macOS
+
+While in offline mode recent versions of macOS does not send out dns queries, and the process for using the local dns server from {project} is a bit more involved then
+other OS.
+
+=== Enable tap devices
+
+Check for the presence of `tapX` (`tap0`, `tap1`, etc..) in `/dev` directory. If no `tap` device is present, you need to install the `tuntap` package.
+
+----
+$ brew install tuntap
+----
+
+=== Use a tap device to create a network service
+
+We need to create a network service using a `tap` device, for this we need to patch the file `/Library/Preferences/SystemConfiguration/preferences.plist`.
+Open the file using the root user or with sudo permissions and add the following section of *xml* under the element *<key>NetworkServices</key>*:
+
+----
+<key>D16F22CE-6DDE-4E63-837C-E16538EA5CCB</key>	//<1>
+<dict>
+    <key>DNS</key>
+    <dict />
+    <key>IPv4</key>
+    <dict>
+        <key>Addresses</key>
+        <array>
+            <string>10.10.90.1</string>		//<2>
+        </array>
+        <key>ConfigMethod</key>
+        <string>Manual</string>
+        <key>SubnetMasks</key>
+        <array>
+            <string>255.255.0.0</string>
+        </array>
+    </dict>
+    <key>IPv6</key>
+    <dict>
+        <key>ConfigMethod</key>
+        <string>Automatic</string>
+    </dict>
+    <key>Interface</key>
+    <dict>
+        <key>DeviceName</key>
+        <string>tap0</string>			//<3>
+        <key>Hardware</key>
+        <string>Ethernet</string>
+        <key>Type</key>
+        <string>Ethernet</string>
+        <key>UserDefinedName</key>
+        <string>MiniTap</string>		//<4>
+    </dict>
+    <key>Proxies</key>
+    <dict>
+        <key>ExceptionsList</key>
+        <array>
+            <string>*.local</string>
+            <string>169.254/16</string>
+        </array>
+        <key>FTPPassive</key>
+        <integer>1</integer>
+    </dict>
+    <key>SMB</key>
+    <dict />
+    <key>UserDefinedName</key>
+    <string>MiniTap</string>			//<5>
+</dict>
+----
+
+<1> This is the `uuid` for the network service, use `uuidgen` to generate and replace it.
+<2> The IP address for the network service.
+<3> Tap device to use, in the above example it is `tap0` present in `/dev`.
+<4> Name for the network service (This will appear in the GUI).
+<5> Same as //<4>.
+
+=== Adding the Network Service to _ServiceOrder_ array
+
+Look for the element *<key>ServiceOrder</key>* in the same file, and append the uuid for our _MiniTap_ network service to the array.
```

Comment:
`<key>ServiceOrder</key>` should be styled with backticks. UUID should be capitalized. Also, we shouldn't assume that the user is following along linearly -- they may be coming back to this documentation after some time, so let's name the file they need to look in.

```adoc
In the *_/Library/Preferences/SystemConfiguration/preferences.plist_* file, look for the `<key>ServiceOrder</key>` element.
As root, append the UUID for our *MiniTap* network service to this array. 
```